stdlib/event/time.lua

```diff
@@ -0,0 +1,73 @@
+--- Time Event module
+-- @module Event.Time
+
+require 'stdlib/event/event'
+
+Event.Time = {}
+Event.Time._last_change = {}
+
+--All times are offset by 0.5
+--This is because both EvoGUI and MoWeather already apply that offset.
+--Following the precedent to remain consistent.
+--Actually, this little snippet is even borrowed from EvoGUI.
+if remote.interfaces.MoWeather then
+    -- assume MoWeather's getdaytime is sane
+    function Event.Time.get_day_time(surface_name_or_index) return remote.call("MoWeather", "getdaytime", surface_name_or_index) end
+else
+    -- 0.5 is midnight; let's make days *start* at midnight instead.
+    function Event.Time.get_day_time(surface_name_or_index) return game.surfaces[surface_name_or_index].daytime + 0.5 end
+end
+
+--These are Surface only events
+--- @field Gets around ldoc V1.4.3 eating the first field. :(
+Event.Time.ldoc_V1_4_3_eats_this   = nil
+--- @field Fires whenever it becomes midday/noon on a surface
+Event.Time.midday   = script.generate_event_name()
+--- @field Fires whenever it becomes midnight on a surface
+Event.Time.midnight = script.generate_event_name()
+--- @field Fires whenever the sunrises on a surface
+Event.Time.sunrise  = script.generate_event_name()
+--- @field Fires whenever the sunsets on a surface
+Event.Time.sunset   = script.generate_event_name()
+--- @field Fires every hour for a surface
+Event.Time.hourly   = script.generate_event_name()
+--- @field Fires every day for a surface
+Event.Time.daily    = script.generate_event_name()
+
+Event.register(defines.events.on_tick, function(event)
+    for _, surface in pairs(game.surfaces) do
+        local day_time = math.fmod(Event.Time.get_day_time(surface.index), 1)
+        local day_time_minutes = math.floor(day_time * 24 * 60)
+
+        if day_time_minutes ~= Event.Time._last_change[surface.name] then
+            Event.Time._last_change[surface.name] = day_time_minutes
+
+            if day_time_minutes % 60 == 0 then
+                game.raise_event(Event.Time.hourly, {surface = surface})
```

Comment:
That is what I thought too but in all the testing I did game.raise_event /only/ seems to raise the event for the mod that called the function. Originally I had the time events add to defines.events.whatever. the 4 test mods I had all saw the stdlib time events and the 3 "slow" mods would not register their on_tick event to prevent extra time events. However only the mod that called game.raise_event would actually *see* the event. the other 3 mods despite listening for the same event with the same id number would never see the event.

Feel free to try it yourself and see if I was screwing it up because it would make more sense if all mods would see events from game.raise_event instead of just the mod that actually did the call to the function.